enhancements/arbiter-clusters.md

```diff
@@ -0,0 +1,628 @@
+compute:
+  - architecture: amd64
+    hyperthreading: Enabled
+    name: worker
+    platform: {}
+    replicas: 0
+arbiterNode:
+  architecture: amd64
+  hyperthreading: Enabled
+  replicas: 1
+  name: arbiter
+  platform:
+    aws:
+      type: m6a.large
+controlPlane:
+  architecture: amd64
+  hyperthreading: Enabled
+  name: master
+  platform:
+    aws:
+      type: m6a.4xlarge
+  replicas: 2
+```
+
+Ex. `/machines/arbiter.go`
+
+```go
+type Arbiter struct {
+	UserDataFile           *asset.File
+	MachineConfigFiles     []*asset.File
+	MachineFiles           []*asset.File
+	ControlPlaneMachineSet *asset.File
+	IPClaimFiles           []*asset.File
+	IPAddrFiles            []*asset.File
+
+	// SecretFiles is used by the baremetal platform to register the
+	// credential information for communicating with management
+	// controllers on hosts.
+	SecretFiles []*asset.File
+
+	// NetworkConfigSecretFiles is used by the baremetal platform to
+	// store the networking configuration per host
+	NetworkConfigSecretFiles []*asset.File
+
+	// HostFiles is the list of baremetal hosts provided in the
+	// installer configuration.
+	HostFiles []*asset.File
+}
+```
+
+### MCO Changes
+
+Currently MCO blocks machine config pools that target non master/worker roles in
+a [validation
+webhook](https://github.com/openshift/machine-config-operator/blob/7c5ae75515fe373e3eecee711c18b07d01b5e759/manifests/machineconfigcontroller/custom-machine-config-pool-selector-validatingadmissionpolicy.yaml#L20-L25).
+We will need to update this to allow the new arbiter role.
+
+`custom-machine-config-pool-selector-validatingadmissionpolicy.yaml`
+
+```
+          has(object.spec.machineConfigSelector.matchLabels) &&
+          (
+            (object.spec.machineConfigSelector.matchLabels["machineconfiguration.openshift.io/role"] == "master")
+            ||
+            (object.spec.machineConfigSelector.matchLabels["machineconfiguration.openshift.io/role"] == "worker")
+            ||
+            (object.spec.machineConfigSelector.matchLabels["machineconfiguration.openshift.io/role"] == "arbiter")
+          )
+```
+
+We will also need to update the go logic in the renderer to allow the `arbiter`
+role.
+
+`render.go`
+
+```go
+// GenerateMachineConfigsForRole creates MachineConfigs for the role provided
+func GenerateMachineConfigsForRole(config *RenderConfig, role, templateDir string) ([]*mcfgv1.MachineConfig, error) {
+	rolePath := role
+	//nolint:goconst
+	if role != "worker" && role != "master" && role != "arbiter" {
+		// custom pools are only allowed to be worker's children
+		// and can reuse the worker templates
+		rolePath = "worker"
+	}
+...
+```
+
+We then need to add new rendered initial files for the arbiter role so
+bootstrapping occurs correctly. When MCO stands up the Machine Config Server
+today, we supply initial configs from the
+[template/<master/worker>/00-<master/worker>](https://github.com/openshift/machine-config-operator/tree/master/templates)
```

Comment:
One thing here is I'd like to make it such that any cluster that isn't using arbiter nodes to not have an arbiter pool. As it stands I think the proposal will make it such that we always create an arbiter pool, just empty and with no nodes. Since it almost looks like a custom pool, I'd prefer we don't confuse users into thinking we generated a new pool for them.

Reply:
That's a great call out, I'll add that wording to make sure we only create this pool when the arbiter is specified